cmd/vic-machine/ops-user/rbac.go

```diff
@@ -0,0 +1,286 @@
+// Copyright 2016-2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opsuser
+
+import (
+	"context"
+
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/types"
+	"github.com/vmware/vic/lib/config"
+	"github.com/vmware/vic/pkg/vsphere/session"
+)
+
+const (
+	VCenter = iota
+	Datacenter
+	Cluster
+	VSANDatastore
+	DatastoreFolder
+	DataStore
+	Network
+	Endpoint
+)
+
+type AuthzManager struct {
+	authzManager *object.AuthorizationManager
+	session      *session.Session
+	vchConfig    *config.VirtualContainerHostConfigSpec
+	principal    string
+}
+
+type Resource struct {
+	Type      int8
+	Propagate bool
+	Role      types.AuthorizationRole
+}
+
+type RBACConfig struct {
+	Resources []Resource
+}
+
+type PermissionList []types.Permission
+
+var rolePrefix = "vch-"
```

Comment:
Can we make this `vic-vch-` - while VCH is a very common acronym to us VIC is the name of the product that most people seem to use.